src/Symfony/Component/Security/Core/Authorization/AccessDecisionManager.php

```diff
@@ -58,9 +56,13 @@ public function __construct(array $voters = array(), $strategy = self::STRATEGY_
      * Configures the voters.
      *
      * @param VoterInterface[] $voters An array of VoterInterface instances
+     *
+     * @deprecated since version 3.3, to be removed in 4.0. Pass the voters to the constructor instead.
      */
     public function setVoters(array $voters)
     {
+        @trigger_error(sprintf('The %s() method is deprecated since version 3.3 and will be removed in 4.0. Pass the voters to the constructor instead.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
Changing the type of `$voters` from `array` to `iterable` would be a BC break so this would mean having a different argument type for the constructor parameter and this method.